### ISSUE TITLE
Custom Data not updating when calling save() on resource

### ISSUE THREAD

**Issue by ecrisostomo** [stormpath/stormpath-sdk-php#47]
The custom data does not get updated on Stormpath when calling the parent resource's save() method.

For example, the following code doesn't save the custom data on the account, but it does save the middleName property:

```php
$account->setMiddleName('Test middle name'); 
$customData = $account->customData; 
$customData->companyName = 'Company Test'; 
$account->save();
```

In order to save the customData, the save() method would need to be called on the customData object which shouldn't be the case since the REST API allows the update on the custom data when updating the resource.

**Comment by omgitstom** [comment]
Elder notes:  Make sure we are using the dirty properties correctly for customData

**Comment by omgitstom** [comment]
May be a dupe of #42

**Comment by omgitstom** [comment]
Fixed in 1.5
